pkg/database/sql/reader.go

```diff
@@ -20,6 +20,23 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+// convertWildcardToSQL converts user wildcard patterns to SQL LIKE patterns
+// * becomes % (matches any sequence of characters)
+// Converts to lowercase for case-insensitive matching
+func convertWildcardToSQL(pattern string) string {
+	if pattern == "" {
+		return pattern
+	}
+	// Convert wildcard pattern to SQL LIKE pattern and make it case-insensitive
+	sqlPattern := strings.ReplaceAll(pattern, "*", "%")
+	return strings.ToLower(sqlPattern)
+}
+
+// hasWildcards checks if a name pattern contains wildcard characters
+func hasWildcards(pattern string) bool {
+	return strings.Contains(pattern, "*")
+}
```

Comment:
Probably `strings.Contains(pattern, "*")` was not that hard to read to justify a function.